client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-large-preview.tsx

```diff
@@ -253,7 +285,21 @@ const PatternLargePreview = ( {
 		return () => {
 			frameRef.current?.removeEventListener( 'mouseleave', handleMouseLeave );
 		};
-	}, [ frameRef, hasSelectedPattern, setActiveElement ] );
+	}, [ frameRef, hasSelectedPattern, setActiveElement, setShouldShowTooltip ] );
+
+	// Tooltip follows the mouse cursor.
+	useEffect( () => {
+		const handleMouseMove = ( event: MouseEvent ) => {
+			if ( shouldShowTooltip ) {
+				setTooltipStyles( { left: event.clientX, top: event.clientY } );
```

Comment:
I think we can keep the anchor starting from (0, 0), and set the styles to the tooltip element here. Then, we can avoid the component re-rendering as we don't need to call `setState` anymore. WDYT?

```js
tooltipRef.current.style.transform = `translate( ${event.clientX}px, ${event.clientY}px )`;
```

Reply:
Sounds good! That said, if we use this approach, then we probably won't be able to use the benefits of the `Popover` component since the anchor will be at `0, 0`. On the other hand, `Popover` functionalities such as `flip` already doesn't work, so I wonder if we need to use this component at all. WDYT?

Reply:
I'm not a fan of the `Popover` component. 

Side note: I'm considering removing the tooltip completely from pattern previews to avoid the dancing if I cannot find a solution to show it only on `:focus-visible`. Sharing this note in case you know how to do it. p1697631268108279-slack-CRWCHQGUB

Reply:
I looked at that issue before and IIRC found that the show-on-focus behavior is built-in to the Popover component so it's hard to override 🙁